File: backend/src/routes/account_routes.py
# ====================================================================
# 🎛️ RUTAS DE CUENTAS
# ====================================================================

# Importamos las librerías necesarias
from flask import Blueprint
from src.controllers.account_controller import (create_account, get_user_accounts)

# * Definir Blueprint para las rutas de cuentas
account_bp = Blueprint('account_bp', __name__)

# ====================================================================
# 🆕 Ruta para crear una cuenta
# ====================================================================
@account_bp.route('/create', methods=['POST'])
def create_account_route():
    """
    Ruta para crear una nueva cuenta.
    """
    return create_account()

# ====================================================================
# 🔎 Ruta para obtener todas las cuentas de un usuario
# ====================================================================
@account_bp.route('/', methods=['GET'])
def get_user_accounts_route():
    """
    Ruta para obtener todas las cuentas de un usuario autenticado.
    """
    return get_user_accounts()

File: backend/src/routes/accountRoutes.js
//###################################################
// 🪪🌐 Rutas de Cuentas Bancarias
//###################################################

const express = require("express");
const router = express.Router();
const accountController = require("../controllers/accountController");
const authMiddleware = require("../middlewares/authMiddleware");

// * Crear una nueva cuenta
router.post("/", authMiddleware, accountController.createAccount);

// ? Obtener todas las cuentas del usuario autenticado
router.get("/", authMiddleware, accountController.getAccounts);

// ? Obtener una cuenta por ID
router.get("/:id", authMiddleware, accountController.getAccountById);

// ? Actualizar saldo o información de la cuenta
router.put("/:id", authMiddleware, accountController.updateAccount);

// ! Eliminar una cuenta
router.delete("/:id", authMiddleware, accountController.deleteAccount);

module.exports = router;
